packages/integrations/src/base/factory.ts

```diff
@@ -0,0 +1,16 @@
+import type { IntegrationKind } from "@homarr/definitions";
+
+import { HomeAssistantIntegration } from "../homeassistant/homeassistant-integration";
+import { PiHoleIntegration } from "../pi-hole/pi-hole-integration";
+import type { IntegrationInput } from "./integration";
+
+export const integrationFactory = (kind: IntegrationKind, integration: IntegrationInput) => {
```

Comment:
**NIT**
I dislike the name of this -> It is true that this function is a factory but I would expect this name to see as the class name. The process of instantiating is not called "integrationFactory".